sdk/core/Azure.Core/tests/RequestContextTests.cs

```diff
@@ -27,5 +34,261 @@ public void CanSetErrorOptions()
+            Assert.AreEqual("RequestContext", perRetryValues.ElementAt(2));
+
+            Assert.IsTrue(request.Headers.TryGetValues("BeforeTransport", out var beforeTransportValues));
+            Assert.AreEqual(2, beforeTransportValues.Count());
+            Assert.AreEqual("ClientOptions", beforeTransportValues.ElementAt(0));
+            Assert.AreEqual("RequestContext", beforeTransportValues.ElementAt(1));
+        }
+
+        public class AddHeaderPolicy : HttpPipelineSynchronousPolicy
+        {
+            private string _headerName;
+            private string _headerVaue;
+
+            public AddHeaderPolicy(string headerName, string headerValue) : base()
+            {
+                _headerName = headerName;
+                _headerVaue = headerValue;
+            }
+
+            public override void OnSendingRequest(HttpMessage message)
+            {
+                message.Request.Headers.Add(_headerName, _headerVaue);
+            }
+        }
+
+        [Test]
+        [NonParallelizable]
+        public async Task CanSuppressLoggingAsError()
+        {
+            // logging setup
+            var listener = new TestEventListener();
+            listener.EnableEvents(AzureCoreEventSource.Singleton, EventLevel.Verbose);
+
+            // pipeline setup
+            var response = new MockResponse(404);
+            var mockTransport = new MockTransport(response);
+
+            var pipeline = new HttpPipeline(mockTransport, new[] { new LoggingPolicy(logContent: true, int.MaxValue, HttpMessageSanitizer.Default, "Test SDK") });
+            var context = new RequestContext();
+            context.AddClassifier(new int[] { 404 }, isError: false);
+            var message = pipeline.CreateMessage(context);
+
+            await pipeline.SendAsync(message, context.CancellationToken);
+
+            Assert.IsFalse(message.Response.IsError);
+
+            EventWrittenEventArgs e = listener.SingleEventById(5); // ResponseEvent
+            Assert.AreEqual(EventLevel.Informational, e.Level);
+            Assert.AreNotEqual(EventLevel.Warning, e.Level);
+            Assert.AreNotEqual(EventLevel.Error, e.Level);
+            Assert.AreEqual("Response", e.EventName);
+            Assert.AreEqual(e.GetProperty<int>("status"), 404);
+        }
+
+        [Test]
+        [NonParallelizable]
+        public async Task CanSuppressTracingAsError()
+        {
+            Activity activity = null;
+            using var testListener = new TestDiagnosticListener("Azure.Core");
+
+            // pipeline setup
+            MockTransport mockTransport = new MockTransport(_ =>
+            {
+                activity = Activity.Current;
+                MockResponse mockResponse = new MockResponse(409);
+                return mockResponse;
+            });
+
+            var pipeline = new HttpPipeline(mockTransport, new[] { new RequestActivityPolicy(true, "Microsoft.Azure.Core.Cool.Tests", HttpMessageSanitizer.Default) });
```

Comment:
Microsoft.Azure.Core.Cool.Tests
![](https://media2.giphy.com/media/62PP2yEIAZF6g/giphy.gif)

Reply:
Oh that is a Pavelism I meant to edit out!  Will fix.